pkg/apis/pmemcsi/v1alpha1/deployment_types.go

```diff
@@ -109,13 +110,76 @@ type DeploymentSpec struct {
 	KubeletDir string `json:"kubeletDir,omitempty"`
 }
 
+// DeploymentConditionType type for representing a deployment status condition
+type DeploymentConditionType string
+
+const (
+	// CertsVerified means the provided deployment secrets are verified and valid for usage
+	CertsVerified DeploymentConditionType = "CertsVerified"
+	// CertsReady means secrests/certificates required for running the PMEM-CSI driver
+	// are ready and the deployment could progress further
+	CertsReady DeploymentConditionType = "CertsReady"
+	// DriverDeployed means that the all the sub-resources required for the deployment CR
+	// got created
+	DriverDeployed DeploymentConditionType = "DriverDeployed"
+)
+
+// +k8s:deepcopy-gen=true
+type DeploymentCondition struct {
+	// Type of condition.
+	Type DeploymentConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status"`
+	// Message human readable text that explain why this condition is in this state
+	// +optional
+	Reason string `json:"reason,omitempty"`
+	// Last time the condition was probed.
+	// +optional
+	LastUpdateTime metav1.Time `json:"lastUpdateTime,omitempty"`
+}
+
+type DriverType int
+
+const (
+	ControllerDriver DriverType = iota
+	NodeDriver
+)
+
+func (t DriverType) String() string {
+	switch t {
+	case ControllerDriver:
+		return "controller"
+	case NodeDriver:
+		return "node"
+	}
+	return ""
+}
+
+// +k8s:deepcopy-gen=true
+type DriverStatus struct {
+	// Type represents type of the driver: controller or node
+	Type string `json:"type"`
+	// Status represents the driver status : Ready, NotReady
+	Status string `json:"status"`
```

Comment:
Please document here what "ready" means for the driver status.

Reply:
Add the description.